playground/waitfor/WaitForSandbox.AppHost/Program.cs

```diff
@@ -35,3 +38,32 @@
 #endif
 
 builder.Build().Run();
+
+public static class PgDumpExtensions
+{
+    public static IResourceBuilder<PostgresServerResource> WithPgDumpCommands(this IResourceBuilder<PostgresServerResource> builder)
+    {
+        builder.ApplicationBuilder.Eventing.Subscribe<BeforeStartEvent>((e, ct) =>
+        {
+            var executor = e.Services.GetRequiredService<ContainerCommandExecutor>();
+
+            var databases = e.Model.Resources.OfType<IResourceWithParent<PostgresServerResource>>().Where(r => r.Parent == builder.Resource).ToList();
+            foreach (var db in databases)
+            {
+                var dbBuilder = builder.ApplicationBuilder.CreateResourceBuilder(db);
```

Comment:
This super late model of mutating the resource builder is pretty broken... you know how I feel about this.

Reply:
This is BeforeStart. Not late at all.